kolibri_oidc_client_plugin/auth.py

```diff
@@ -68,10 +69,22 @@ def create_user(self, claims):
         if not full_name:
             full_name = '{} {}'.format(claims.get('given_name', ""), claims.get('family_name', ""))
         # not needed in Kolibri, email is not mandatory:
-        email = username
+        email = claims.get("email", username)
         # Kolibri doesn't allow an empty password. This isn't going to be used:
         password = uuid4().hex
-
-        return self.UserModel.objects.create_user(
-            username, email=email, full_name=full_name, password=password
+        birthdate = claims.get("birthdate", "NOT_SPECIFIED")
+        gender = claims.get("gender", "NOT_SPECIFIED")
+        user = self.UserModel.objects.create_user(
+            username, email=email, full_name=full_name, password=password,
+            birth_year=birthdate, gender=gender
         )
+
+        # check if the user has assigned roles and assign them in such case
+        roles = claims.get("roles", [])
+        for role in roles:
+            try:
+                user.facility.add_role(user, role.lower())
+            except InvalidRoleKind:
+                pass  # The role does not exist in Kolibri
```

Comment:
maybe log this error?

Reply:
> This looks good.
> 
> Main feedback is that the fields provided don't map perfectly to our internal models. For example birthday vs year, first/last vs full name, no support for email. This mapping would need to be defined and documented

Actually, that's one of the task of this code, field  names are defined in [the standard](https://openid.net/specs/openid-connect-core-1_0.html#StandardClaims) and this code retrieves them and insert them into our model with the required changes.

Reply:
> with the required changes

The input fields might be standardized, but we still need to map them to Kolibri (like you already did with full name, by concatenating two of the input fields).

For example:

* birth date is not the same as birth year because some information is lost
* email should not be documented at all, or else we might want to map it to the `identifier` field because email is not a supported field in Kolibri
* I believe gender currently takes a specific set of enumerated values

Reply:
ah, I got what you mean, I have fixed gender an birth date. For email, I see this in our code https://github.com/learningequality/kolibri/blob/develop/kolibri/core/auth/models.py#L572 so, what's the problem of adding it for kolibri to handle it as it wishes?

Reply:
Done